.github/workflows/generator.yaml

```diff
@@ -19,32 +19,20 @@ jobs:
   generation:
     name: Generated artifacts
     runs-on: ubuntu-20.04
+    container: 
+      image: swiftnav/libsbp-build:latest-master
+      options: --user root
+      volumes:
+        - ${{ github.workspace }}:/mnt/workspace
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           fetch-depth: 0
           submodules: recursive
 
-      - name: Installing dependencies
+      - name: Set safe git directory
         run: |
-          sudo apt-get -qq update
-          sudo apt-get install -y tox python3-sphinx clang-format-6.0 enchant
-          pip3 install tox-run-command
-          tox -e py --notest
-
-      - name: Installing Node
-        uses: actions/setup-node@v4
-        with:
-          node-version: 18
-
-      - name: Installing Quicktype
-        run: |
-          npm ci
-          npm install -g quicktype
-
-      - uses: gradle/gradle-build-action@v2
-        with:
-          gradle-version: 7.1.1
+          git config --global --add safe.directory /__w/libsbp/libsbp
```

Comment:
Not sure this is ideal but required to resolve a permissions issue which I think is caused by different UIDs being used inside and outside the container

Reply:
I think this is fine/I don't know a better solution here.